src/NuGet.Clients/NuGet.SolutionRestoreManager/VulnerablePackagesInfoBar.cs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Telemetry;
+
+#nullable enable
+
+namespace NuGet.SolutionRestoreManager
+{
+    [Export(typeof(IInfoBarService))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class VulnerablePackagesInfoBar : IInfoBarService, IVsInfoBarUIEvents
+    {
+        IAsyncServiceProvider _asyncServiceProvider = AsyncServiceProvider.GlobalProvider;
+        private IVsInfoBarUIElement? _infoBarUIElement;
+        private bool _visible;
+        private bool _closed = false;
+        private bool _closeFromHide = false;
+        private uint? _eventCookie;
+
+        [Import]
+        private Lazy<IPMUIStarter>? PackageManagerUIStarter { get; set; }
+
+        public async Task ShowAsync(CancellationToken cancellationToken)
+        {
+            await CreateAndShowInfoBarAsync(cancellationToken);
+        }
+
+        public async Task RemoveAsync(CancellationToken cancellationToken)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            if (!_visible)
+            {
+                return;
+            }
+
+            _closeFromHide = true;
+
+            try
+            {
+                if (_eventCookie.HasValue)
+                {
+                    _infoBarUIElement?.Unadvise(_eventCookie.Value);
+                    _infoBarUIElement?.Close();
+                    _infoBarUIElement = null;
+                }
+            }
+            finally
+            {
+                _closeFromHide = false;
+            }
+        }
+
+        protected InfoBarModel GetInfoBarModel()
+        {
+            return new InfoBarModel(
+                Resources.InfoBar_TextMessage,
+                new IVsInfoBarActionItem[] {
+                    new InfoBarHyperlink(Resources.InfoBar_HyperlinkMessage),
+                },
+                KnownMonikers.StatusWarning);
+        }
+
+        internal async Task CreateAndShowInfoBarAsync(CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            if (_closed || _visible)
```

Comment:
This is probably related to your comment here: https://github.com/NuGet/NuGet.Client/pull/5258#issuecomment-1629477890.

If so, what happens if the user loads a new solution? Do we want to suppress all info bar warnings for the remainder of the VS session on all solutions because they closed the infobar on one solution? Or would we want to reset the suppression of the infobar and show it for the newly loaded solution?

Reply:
Yeah thats related to the DevDiv design Guide, 

> All infobars will have a “×” close button in the far-right to close the message. The message won’t show again during that session once it’s been closed.

Personally, I would like to show the infobar every time you open a Solution but I think we must follow the guidelines here 😢